ai-compile/vscode-extension/src/pymacer.ts

```diff
@@ -1,80 +1,108 @@
+/* To better understand/ appreciate the various interfaces declared here,
+ * please refer MACER paper at https://arxiv.org/abs/2005.14015
+*/
+
 import * as vscode from "vscode";
 import { promisify } from "util";
-import { exec, spawn } from "child_process";
-
+import { exec } from "child_process";
 import axios from "axios";
 import * as https from "https";
-import * as fs from "fs";
 
+// request to backend server
 export interface Payload {
   source: string;
   lastEditLine: number;
 }
 
 export interface Edit {
-  type: string;
-  start: number;
-  end: number;
-  insertString: string;
+  type: string;           // type of edit operation - insert/ delete/ replace
+  start: number;          // starting position of the edit to be made
+  end: number;            // ending position of the edit to be made
+  insertString: string;   // text to be inserted in the range defined above
 }
 
 export interface Feedback {
-  fullText: string;
-  msg1: string;
-  msg2: string;
+  fullText: string;       // complete Feedback Text 
+  // partial Feedback Texts
+  msg1: string;           // generic message quoting existance of error
+  msg2: string;           // specific message quoting kind of edits required to correct
   misc: string;
-  actionMsg: string;
-  action: string;
-  tokens: string[];
-  tokensText: string[];
+  // relates to the type of edit - insert/ delete/ replace
+  actionMsg: string;    
+  action: string;       
+  tokens: string[];       // actual tokens on which the above action has to be performed
+  tokensText: string[];   // textual description of the above tokens
 }
 
+// response from backend server
 export interface Response {
   lineNo: number;
-  repairLine: string;
+  repairLine: string;     // corrected version of the erroneous source line
   editDiffs: Edit[];
   feedback: Feedback[];
+  // representation the repair involved in making a fix.
   repairClasses: string[];
 }
 
 export type Fixes = Response[] | undefined;
 
 export interface DocumentStore {
-  // TODO: Can this be made private? Does it make sense?
   filePath: string;
   fixes: Fixes;
 }
 
-export const baseURL: string | undefined = vscode.workspace
+export class Document implements DocumentStore {
+  private _filePath: string;
+  private _fixes: Fixes;
+
+  constructor(filePath: string, fixes: Fixes) {
+    this._filePath = filePath;
+    this._fixes = fixes;
+  }
+
+  public get fixes() {
+    return this._fixes;
+  }
+  public set fixes(value: Fixes) {
+    this._fixes = value;
+  }
+
+  public get filePath() {
+    return this._filePath;
+  }
+  public set filePath(value: string) {
+    this._filePath = value;
+  }
+}
+
+export const serverURL: string | undefined = vscode.workspace
   .getConfiguration("python-hints")
   .get("webServerPath");
 
 export const requestTimeOut: number = 100000;
```

Comment:
Have a new file called `Config` and pull all these hard-coded numbers from there. Again, never hardcode any constant within the source code.